JASP-Engine/JASP/R/mediationanalysis.R

```diff
@@ -0,0 +1,573 @@
+        v1 <- .v(options$predictor[idx_mat[i,1]])
+        v2 <- .v(options$predictor[idx_mat[i,2]])
+        pred_res_part <- paste0(pred_res_part, "\n", v1, " ~~ ", v2)
+      }
+      pred_res_part <- paste0(pred_res_part, "\n\n")
+    }
+  }
+
+  med_res_part <- NULL
+  if (n_medi > 1) {
+    med_res_part <- "# mediator residual covariance"
+    idx_mat  <- which(upper.tri(diag(n_medi)), arr.ind = TRUE)
+    for (i in 1:nrow(idx_mat)) {
+      v1 <- .v(options$mediators[idx_mat[i,1]])
+      v2 <- .v(options$mediators[idx_mat[i,2]])
+      med_res_part <- paste0(med_res_part, "\n", v1, " ~~ ", v2)
+    }
+    med_res_part <- paste0(med_res_part, "\n\n")
+  }
+
+  res_part <- NULL
+  if (n_deps > 1) {
+    res_part <- "# dependent residual covariance"
+    idx_mat  <- which(upper.tri(diag(n_deps)), arr.ind = TRUE)
+    for (i in 1:nrow(idx_mat)) {
+      v1 <- .v(options$dependent[idx_mat[i,1]])
+      v2 <- .v(options$dependent[idx_mat[i,2]])
+      res_part <- paste0(res_part, "\n", v1, " ~~ ", v2)
+    }
+    res_part <- paste0(res_part, "\n\n")
+  }
+
+
+  dec_part <- "# effect decomposition"
+  for (d in 1:n_deps) {
+    for (p in 1:n_pred) {
+      dec_part <- paste0(dec_part, "\n# y", d, " ~ x", p)
+      parnames <- c()
+      for (m in 1:n_medi) {
+        par_name <- paste0("ind_x", p, "_m", m, "_y", d)
+        parnames <- c(parnames, par_name)
+        a_name   <- paste0("a", m, p)
+        b_name   <- paste0("b", d, m)
+        dec_part <- paste0(dec_part, "\n", par_name, " := ", a_name, "*", b_name)
+      }
+      dec_part <- paste0(dec_part, "\nind_x", p, "_y", d, "    := ", paste(parnames, collapse = " + "))
+      dec_part <- paste0(dec_part, "\ntot_x", p, "_y", d, "    := ", "ind_x", p, "_y", d, " + c", d, p, "\n")
+    }
+  }
+
+  return(paste0(title, dep_part, med_part, conf_part,
+                pred_res_part, med_res_part, res_part,
+                dec_part))
+}
+
+# Output functions ----
+
+.medParTable <- function(jaspResults, medResult, options, errors) {
+  showOpts <- c("showdir", "showind", "showtotind", "showtot", "showres")
+  if (!any(options[showOpts])) return()
+  
+  jaspResults[["parest"]] <- pecont <- createJaspContainer("Parameter estimates")
+  pecont$dependOn(options = c("ciWidth", "bootCItype"), optionsFromObject = jaspResults[["stateMedResult"]])
+  pecont$position <- 0
+
+  if (is.null(medResult)) {
+    pecont[["dir"]] <- dirtab <- createJaspTable(title = "Direct effects")
+    pecont[["ind"]] <- indtab <- createJaspTable(title = "Indirect effects")
+    pecont[["tot"]] <- tottab <- createJaspTable(title = "Total effects")
+
+    dirtab$addColumnInfo(name = "est",    title = "Estimate",   type = "number")
+    dirtab$addColumnInfo(name = "se",     title = "Std. Error", type = "number")
+    dirtab$addColumnInfo(name = "z",      title = "z-value",    type = "number")
+    dirtab$addColumnInfo(name = "pvalue", title = "p",          type = "number")
+    dirtab$setExpectedSize(rows = 1, cols = 4)
```

Comment:
An empty table is automatically of the size [1, ncols] in the output, so this `setExpectedSize` is not necessary

Reply:
Nice!